save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt

```diff
@@ -177,30 +474,130 @@ class ProjectView : RComponent<ProjectProps, ProjectViewState>() {
         }
     }
 
-    private fun RBuilder.runErrorModal() = modal {
+    private fun RBuilder.runErrorEmptyTestTypesModal() = modal {
+        attrs {
+            isOpen = state.isTestTypesEmpty
+            contentLabel = "Empty Test types"
+        }
+        div {
+            h2("h3 mb-0 text-gray-800") {
+                +"Please select at least one type"
+            }
+        }
+        button(type = ButtonType.button, classes = "btn btn-primary") {
+            attrs.onClickFunction = { setState { isTestTypesEmpty = false } }
+            +"Close"
+        }
+    }
+
+    private fun RBuilder.runErrorNotGitUrl() = modal {
+        attrs {
+            isOpen = state.isNoGitUrl
+            contentLabel = "No git url"
+        }
+        div {
+            h2("h3 mb-0 text-gray-800") {
+                +"Please provide a git url"
+            }
+        }
+        button(type = ButtonType.button, classes = "btn btn-primary") {
+            attrs.onClickFunction = { setState { isNoGitUrl = false } }
+            +"Close"
+        }
+    }
+
+    private fun RBuilder.runErrorBothProjectTypesModal() = modal {
```

Comment:
Why can't all these methods be unified as one with error message as an argument?